packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/FIAObjectTranslator.m

```diff
@@ -288,4 +289,86 @@ + (SKPaymentDiscount *)getSKPaymentDiscountFromMap:(NSDictionary *)map
   return discount;
 }
 
++ (nullable SKPaymentTransactionMessage *)convertTransactionToPigeon:
+    (SKPaymentTransaction *)transaction API_AVAILABLE(ios(12.2)) {
+  if (!transaction) {
+    return nil;
+  }
+  SKPaymentTransactionMessage *msg = [SKPaymentTransactionMessage
+            makeWithPayment:[self convertPaymentToPigeon:transaction.payment]
+           transactionState:[self convertTransactionStateToPigeon:transaction.transactionState]
+        originalTransaction:transaction.originalTransaction
+                                ? [self convertTransactionToPigeon:transaction.originalTransaction]
+                                : NULL
+       transactionTimeStamp:[NSNumber numberWithDouble:[transaction.transactionDate
+                                                               timeIntervalSince1970]]
+      transactionIdentifier:transaction.transactionIdentifier
+                      error:[self convertSKErrorToPigeon:transaction.error]];
+  return msg;
+}
+
++ (nullable SKErrorMessage *)convertSKErrorToPigeon:(NSError *)error {
+  SKErrorMessage *msg = [SKErrorMessage makeWithCode:@(error.code)
+                                              domain:error.domain
+                                            userInfo:error.userInfo];
```

Comment:
This may be a problem if the `error.userInfo` contains data that isn't serializable by flutter.  It might be better to keep this empty.

Reply:
sorry, can you explain what you mean? 👀 I think userInfo is at least guaranteed to be a dict

Reply:
`NSDictionary`'s are serializable by Flutter, but the encoder recurses into their contents which can be any NSObject which may not be serializable (see [FlutterStandardCodec.mm](https://github.com/flutter/engine/blob/9142fc4f16a59a0484e1e58eea571a46c11da8c8/shell/platform/darwin/common/framework/Source/FlutterStandardCodec.mm#L254)).  It looks like the behavior is that a "Message corrupted" exception will be thrown on the dart side if that did happen.

If throwing the corrupted exception was the old behavior, I'm fine keeping it.